### PR TITLE
Code Coverage: Added tests for aliases and other complex types

### PR DESCRIPTION
**Pull request by dansteren**
This updates the rust_type_conversions example and adds tests to ensure the syntax stays supported.

### CODE REVIEW

File: examples/rust_type_conversions/src/index.ts [hunk 1]
```diff
@@ -237,60 +240,64 @@ type VariantWithInline = Variant<{
...
-                one: boolean;
-                two: boolean;
-                three: boolean;
-            }>,
-            inline_variant: Variant<{ one: string; two: null; three: boolean }>,
-            inline_func: Func<Query<(param1: string) => string>>
-        ) => string
-    >
->;
+// TODO: Funcs with inline types don't currently work
```
Comment (lastmjs) [inline examples/rust_type_conversions/src/index.ts:243]: Do we have an issue for this?

File: src/compiler/typescript_to_rust/azle_generate/src/ts_ast/ts_type/to_candid_type.rs [hunk 2]
```diff
@@ -21,6 +21,15 @@ impl SourceMapped<'_, TsType> {
             TsType::TsTupleType(_) => {
                 return Err(vec![UnexpectedTsTupleTypes::from_ts_type(self).into()]);
             }
+            TsType::TsArrayType(_) => {
+                // TODO: Because this isn't handled types like this currently break:
```
Comment (lastmjs) [inline src/compiler/typescript_to_rust/azle_generate/src/ts_ast/ts_type/to_candid_type.rs:25]: I don't think this needs to be fixed right? You have to use `Vec`

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: src/compiler/typescript_to_rust/azle_generate/src/ts_ast/ts_type/to_candid_type.rs (hunk 2 above)
Comment (dansteren) [reply src/compiler/typescript_to_rust/azle_generate/src/ts_ast/ts_type/to_candid_type.rs]: Correct. I've deleted this.

File: examples/rust_type_conversions/src/index.ts (hunk 1 above)
Comment (dansteren) [reply examples/rust_type_conversions/src/index.ts:243]: Just made one https://github.com/demergent-labs/azle/issues/1087

**Pull request merged**